public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -115,27 +115,34 @@ export class PanelChrome extends PureComponent<Props, State> {
         return;
       }
 
-      let { errorMessage, isFirstLoad } = this.state;
-
-      if (data.state === LoadingState.Error) {
-        const { error } = data;
-        if (error) {
-          if (errorMessage !== error.message) {
-            errorMessage = error.message;
+      let { isFirstLoad } = this.state;
+      let errorMessage: string | null = null;
+
+      switch (data.state) {
+        case LoadingState.Loading:
```

Comment:
This means that we're mutating `data` somewhere and that's why we get the flickering?

Reply:
no, it only happens on mixed data source or Prometheus that sends responses as each query completes. 

So we do not mutate state really we just send updates to the panel/explore for each query response for these data sources. Works ok in explore but for visualizations that repeat for each series in the dashboard it is not great.

Reply:
Gotcha, thanks